app/controllers/internships_controller.rb

```diff
@@ -1,22 +1,17 @@
 class InternshipsController < ApplicationController
   def index
     @cohort = Cohort.find(params[:cohort_id])
-    @internships = @cohort.internships
     if current_student
-      @internships = @internships.sort_by { |internship| by_student_interest(internship) }
+      @internships = @cohort.internships_sorted_by_interest(current_student)
+    else
+      @internships = @cohort.internships
     end
```

Comment:
take out the `if` here and have `Cohort#internships_sorted_by_interest` return `Cohort#internships` if `current_student` is `nil`.